src/fidesops/service/authentication/authentication_strategy_oauth2.py

```
@@ -0,0 +1,299 @@
+        )
+        # add state to secrets
+        connection_config.secrets["state"] = state
+
+        # assign placeholders in the authorization request config
+        prepared_authorization_request = map_param_values(
+            "authorize",
+            f"{connection_config.name} OAuth2",
+            self.authorization_request,
+            connection_config.secrets,
+        )
+
+        # get the client config from the authorization request or default
+        # to the base client config if one isn't provided
+        client_config = (
+            self.authorization_request.client_config
+            if self.authorization_request.client_config
+            else connection_config.get_saas_config().client_config
+        )
+
+        # build the complete URL with query params
+        return (
+            f"{client_config.protocol}://{assign_placeholders(client_config.host, connection_config.secrets)}"
+            f"{prepared_authorization_request.path}"
+            f"?{urlencode(prepared_authorization_request.query_params)}"
+        )
+
+    def get_access_token(
+        self, db: Session, code: str, connection_config: ConnectionConfig
+    ) -> None:
+        """
+        Generates and executes the access token request based on the OAuth2 config
+        and connection config secrets. The access and refresh tokens returned from
+        this request are stored in the connection config secrets.
+        """
+
+        self._check_required_secrets(connection_config)
+        connection_config.secrets = {**connection_config.secrets, "code": code}
+        access_response = self._call_token_request(
+            "access", self.token_request, connection_config
+        )
+        self._validate_and_store_response(db, access_response, connection_config)
+
+    @staticmethod
+    def _check_required_secrets(connection_config: ConnectionConfig) -> None:
+        secrets = connection_config.secrets
```

Can you either raise an error if secrets are None or do `secrets = connection_config.secrets or {}` ?

---

Changed to ```secrets = connection_config.secrets or {}```